packages/editor-ui/src/views/NodeView.v2.vue

```diff
@@ -625,6 +625,8 @@ function onPinNodes(ids: string[], source: PinDataSource) {
 }
 
 async function onSaveWorkflow() {
+	if (isCanvasReadOnly.value) return;
```

Comment:
could we instead not even emit the `save:workflow` event by adding a `props.readOnly` check [here](https://github.com/n8n-io/n8n/blob/master/packages/editor-ui/src/components/canvas/Canvas.vue#L153) instead?

Reply:
Great suggestion! I'm on it.